src/sema/expression.rs

```diff
@@ -5883,7 +5885,7 @@ fn method_call_pos_args(
             if matches {
                 diagnostics.truncate(marker);
 
-                if !ns.functions[function_no].is_public() {
+                if !ns.functions[*function_no].is_public() {
```

Comment:
@seanyoung We check here if the function is public or external.

Reply:
> > Looks like this routine does not check if the function is declared `public` or `external`.
> > Also the same problem exists for external function calls with named arguments, e.g. `foo.bar({a: 1, b: 2});` (see `method_call_named_args`)
> 
> Can you provide an example of the `foo.bar({a: 1, b: 2})` case?

In the babylink example replace the call to `createPair` with:
```
        uniswapV2Pair = IUniswapV2Factory(_uniswapV2Router.factory())
            .createPair({tokenA: address(this), tokenB:_uniswapV2Router.WETH()});
```
does that work?

Reply:
Indeed, you are right.

Ideally we would give an error if a function is not public, rather than pretending it does not exist. However, that might be a bit much for this PR, going off-topic here

Reply:
We do raise an error here. Take a look at the lines following this comment.

Reply:
Yes. Sorry, I should have paid more attention.